jest/src/main/java/io/searchbox/client/http/JestHttpClient.java

```diff
@@ -126,14 +129,26 @@ protected HttpUriRequest constructHttpMethod(String methodName, String url, Stri
         return httpUriRequest;
     }
 
-    private <T extends JestResult> T deserializeResponse(HttpResponse response, Action<T> clientRequest) throws IOException {
+    private <T extends JestResult> T deserializeResponse(HttpResponse response, final HttpRequest httpRequest, Action<T> clientRequest) throws IOException {
         StatusLine statusLine = response.getStatusLine();
-        return clientRequest.createNewElasticSearchResult(
-                response.getEntity() == null ? null : EntityUtils.toString(response.getEntity()),
-                statusLine.getStatusCode(),
-                statusLine.getReasonPhrase(),
-                gson
-        );
+        try {
+            return clientRequest.createNewElasticSearchResult(
+                    response.getEntity() == null ? null : EntityUtils.toString(response.getEntity()),
+                    statusLine.getStatusCode(),
+                    statusLine.getReasonPhrase(),
+                    gson
+            );
+        } catch (com.google.gson.JsonSyntaxException e) {
+            for (Header header : response.getAllHeaders()) {
+                final String mimeType = header.getValue();
+                if (!mimeType.startsWith("application/json")) {
+                    // probably a proxy that responded in text/html
+                    throw new IOException("Request " + httpRequest.toString() + " yielded " + mimeType
+                            + ", should be json: " + statusLine.toString());
```

Comment:
Adding the original `JsonSyntaxException` as cause here would be good idea eventhough you'd again end up with the long stack trace in your original message. I'd still add it as cause because you never know what kind of edge cases people run into and any information is better than no information at all. :)